waspc/packages/deploy/src/providers/railway/commands/setup/setup.ts

```diff
@@ -0,0 +1,197 @@
+    projectName,
+    options,
+  );
+
+  const project = await setupRailwayProjectForDirectory({
+    projectName,
+    waspProjectDir: options.waspProjectDir,
+    railwayExe: options.railwayExe,
+    existingProjectId: options.existingProjectId,
+  });
+
+  await ensureWaspProjectIsBuilt(options);
+
+  if (project.doesServiceExist(deploymentInstructions.dbServiceName)) {
+    waspSays("Postgres service already exists. Skipping database setup.");
+  } else {
+    await setupDb(deploymentInstructions);
+  }
+
+  if (project.doesServiceExist(deploymentInstructions.clientServiceName)) {
+    waspSays("Client service already exists. Skipping client setup.");
+  } else {
+    await setupClient(deploymentInstructions);
+  }
+
+  if (project.doesServiceExist(deploymentInstructions.serverServiceName)) {
+    waspSays("Server service already exists. Skipping server setup.");
+  } else {
+    await setupServer(deploymentInstructions);
+  }
+}
+
+async function setupRailwayProjectForDirectory({
+  railwayExe,
+  projectName,
+  waspProjectDir,
+  existingProjectId,
+}: {
+  railwayExe: RailwayCliExe;
+  projectName: RailwayProjectName;
+  waspProjectDir: WaspProjectDir;
+  existingProjectId: RailwayProjectId | null;
+}): Promise<RailwayProject> {
+  const { status, project } = await getRailwayProjectStatus({
+    projectName,
+    waspProjectDir,
+    railwayExe,
+    existingProjectId,
+  });
+
+  switch (status) {
+    case ProjectStatus.EXISTING_PROJECT_ALREADY_LINKED:
+      waspSays(
+        `Project with name "${projectName}" already linked. Skipping project creation.`,
+      );
+      return project;
+
+    case ProjectStatus.EXISTING_PROJECT_SHOULD_BE_LINKED:
+      waspSays(`Linking project with name "${project.name}" to this directory`);
+      return linkRailwayProjectToWaspProjectDir(project, {
+        railwayExe,
+        waspProjectDir,
+      });
+
+    case ProjectStatus.MISSING_PROJECT:
+      waspSays(`Setting up Railway project with name "${projectName}"`);
+      return initRailwayProject({
+        projectName,
+        railwayExe,
+        waspProjectDir,
+      });
+
+    default:
+      status satisfies never;
+      throw new Error(`Unhandled status: ${status}`);
+  }
+}
+
+async function setupDb({
+  cmdOptions: options,
+}: DeploymentInstructions<SetupCmdOptions>): Promise<void> {
+  waspSays("Setting up database");
+
+  const railwayCli = createCommandWithCwd(
+    options.railwayExe,
+    options.waspProjectDir,
+  );
+  await railwayCli(["add", "-d", "postgres"]);
+}
+
+async function setupServer({
+  cmdOptions: options,
+  serverServiceName,
+  clientServiceName,
+  dbServiceName,
+}: DeploymentInstructions<SetupCmdOptions>): Promise<void> {
+  waspSays(`Setting up server app with name ${serverServiceName}`);
+
+  // The client service needs a URL so it can be referenced in the
+  // server service env variables.
+  await generateServiceUrl(clientServiceName, clientAppPort, options);
```

Comment:
Aha intersing that this returns nothing, so this has side effect? Creates it up there somwhere, on Railway? So this assumes stuff -> that client service already exists? M aybe let's dcument that? Or even better if we could encode it in type system, but if we can't ok.